woocommerce/woocommerce-android
# Product details: Decouple image gallery from product details

This PR decouples the image gallery from the `ProductDetailsViewModel` so that it can be reused in Variation details. The image gallery functionality should not change.

**To test:**
1. Go to Products -> Tap on a product item
2. Smoke test:
  - adding new images
  - removing images
  - discarding current uploads
3. Verify the update button is displayed correctly when there are changes
